examples/MenuExample.swift

```diff
@@ -0,0 +1,83 @@
+/*
+ Copyright 2017-present The Material Motion Authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import UIKit
+import Transitioning
+
+// This example demonstrates the minimal path to building a custom transition using the Material
+// Motion Transitioning APIs in Swift. The essential steps have been documented below.
+
+class MenuExampleViewController: ExampleViewController {
+
+  func didTap() {
+    let modalViewController = ModalViewController()
+    modalViewController.modalPresentationStyle = .overCurrentContext
```

Comment:
I'm not certain we should require the view controller to be aware that the transition requires a specific presentation style, so I think it would be preferable for `MenuTransition` to conform to `TransitionWithPresentation`. This will ensure that it is able to specify its default expected modal presentation style.